spec/providers/auth.spec.ts

```diff
@@ -61,90 +61,69 @@ describe('AuthBuilder', () => {
-          createdAt: new Date(Date.UTC(2016, 11, 15, 19, 37, 37, 59)),
-          lastSignedInAt: new Date(Date.UTC(2017, 0, 1)),
         },
-        providerData: [{
-          uid: 'g-abcde12345',
-          email: 'foo@gmail.com',
-          displayName: 'My Google Provider Display Name',
-          photoURL: 'googleusercontent.com/foo.jpg',
-          providerId: 'google.com',
-        }],
       };
+    });
 
+    it('should transform old wire format for UserRecord into v5.0.0 format', () => {
       return Promise.all([
-        expect(cloudFunctionCreate(event)).to.eventually.deep.equal(expectedData),
-        expect(cloudFunctionDelete(event)).to.eventually.deep.equal(expectedData),
+        cloudFunctionCreate(event).then(data => {
+          expect(data.metadata.creationTime).to.equal('2016-12-15T19:37:37.059Z');
+          expect(data.metadata.lastSignInTime).to.equal('2017-01-01T00:00:00.000Z');
+        }),
+        cloudFunctionDelete(event).then(data => {
+          expect(data.metadata.creationTime).to.equal('2016-12-15T19:37:37.059Z');
+          expect(data.metadata.lastSignInTime).to.equal('2017-01-01T00:00:00.000Z');
+        }),
       ]);
     });
 
-    // This isn't expected to happen in production, but if it does we should
-    // handle it gracefully.
-    it('should tolerate missing fields in the payload', () => {
-      const cloudFunction = auth.user().onCreate((ev: Event<firebase.auth.UserRecord>) => ev.data);
+    it('should still retain createdAt and lastSignedIn', () => {
+      return Promise.all([
+        cloudFunctionCreate(event).then(data => {
+          expect(data.metadata.createdAt).to.deep.equal(new Date('2016-12-15T19:37:37.059Z'));
+          expect(data.metadata.lastSignedInAt).to.deep.equal(new Date('2017-01-01T00:00:00.000Z'));
+        }),
+        cloudFunctionDelete(event).then(data => {
+          expect(data.metadata.createdAt).to.deep.equal(new Date('2016-12-15T19:37:37.059Z'));
+          expect(data.metadata.lastSignedInAt).to.deep.equal(new Date('2017-01-01T00:00:00.000Z'));
+        }),
+      ]);
+    });
 
-      let event: Event<firebase.auth.UserRecord> = {
+    it('should handle new wire format if/when there is a change', () => {
+      const newEvent = {
         data: {
-          uid: 'abcde12345',
           metadata: {
-            // TODO(inlined) We'll need to manually parse these!
-            createdAt: new Date(),
-            lastSignedInAt: new Date(),
+            creationTime: '2016-12-15T19:37:37.059Z',
+            lastSignInTime: '2017-01-01T00:00:00.000Z',
           },
-          email: 'nobody@google.com',
-          emailVerified: false,
-          displayName: 'sample user',
-          photoURL: '',
-          disabled: false,
         },
-      } as any;
+      };
 
-      return expect(cloudFunction(event)).to.eventually.deep.equal(event.data);
+      return Promise.all([
+        cloudFunctionCreate(newEvent).then(data => {
```

Comment:
We should also expect that the fields with the old name are still present in this case with the new wire format.